src/ucp/core/ucp_request.h

```diff
@@ -207,8 +210,9 @@ struct ucp_request {
                 } rkey_ptr;
 
                 struct {
-                    ucs_ptr_map_key_t req_id;         /* the send request ID on receiver side */
+                    ucs_ptr_map_key_t sreq_id;        /* the send request ID on receiver side */
                     ucp_request_t     *rreq;          /* pointer to the receive request */
+                    ucs_ptr_map_key_t rreq_id;        /* the receive request ID on receiver side */
```

Comment:
do we need both `rreq` and `rreq_id` now?

Reply:
since there is room for both, we thought to leave it there for now to avoid a look-up in the ptr_map.

Reply:
ok

Reply:
Is there any chance to access the pointer when rreq is released? let's say it's canceled on receiver side but sender is sending ACK.
IMO it looks strange I would keep something one and ID is preferrable

Reply:
if the recv side tries to remove an id of a non-existing rreq then the extract function would return NO_ELEM, isn't that what is expected?
or you mean to say we should remove the rreq from the map when it's canceled?

Reply:
- yes, we have to remove id from map before it's put to the pool or in worse case will be reused by next op
- yes, my concern about `rreq` pointer

Reply:
maybe we should add rreq_id as a filed in the receive request itself (in the 'recv' part)
IMO it makes more sense each request will hold its own id